File: Data Structures & Algorithms/single-number/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    singleNumber(nums) {
        let obj = {}
        for (let num of nums){
            obj[num] = (obj[num] || 0) + 1
        }
        console.log(obj);
    for (let key in obj){
        if (obj[key] == 1){
            return key;
        }
    }
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
 
class Solution {
    public int findDuplicate(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int num : nums) {
            // Count occurrences
            map.put(num, map.getOrDefault(num, 0) + 1);

            // If count > 1 → duplicate found
            if (map.get(num) > 1) {
                return num;
            }
        }

        return -1; // should not reach here if there's always a duplicate
    }
}


